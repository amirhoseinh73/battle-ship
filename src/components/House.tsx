import React from "react"
import { useState } from "react"
import { Pressable, StyleSheet, Text } from "react-native"

const getBackground = function (
  filled: boolean,
  isShip: boolean,
  shipKind: BoardFills | null = null,
  shipDestroyed = false
) {
  // hidden ships not hitted yet
  if (isShip && !filled) {
    if (shipKind === "tiny") return { backgroundColor: "#FFFF00" }
    if (shipKind === "small") return { backgroundColor: "green" }
    if (shipKind === "medium") return { backgroundColor: "orange" }
    if (shipKind === "large") return { backgroundColor: "brown" }
  }

  if (isShip && filled) return { backgroundColor: "lightblue" } // first hit

  // ship destroyed
  if (shipDestroyed) {
    if (shipKind === "tiny") return { backgroundColor: "#FFFF00" }
    if (shipKind === "small") return { backgroundColor: "green" }
    if (shipKind === "medium") return { backgroundColor: "orange" }
    if (shipKind === "large") return { backgroundColor: "brown" }
  }

  if (filled) return { backgroundColor: "red" }
  return { backgroundColor: "#888" }
}

const House = function ({
  isShip = false,
  fillType = null,
  id,
}: {
  isShip: boolean
  fillType: BoardFills | null
  id: string
}) {
  const [isFill, setIsFill] = useState(false)

  const fillHouse = () => {
    if (isFill) return
    setIsFill(true)
  }

  return (
    <Pressable
      style={[styles.house, getBackground(isFill, isShip, fillType, false)]}
      onPress={fillHouse}>
      <Text>{id}</Text>
    </Pressable>
  )
}

const styles = StyleSheet.create({
  house: {
    width: "9%",
    height: 32,
    aspectRatio: 1,
    borderRadius: 3,
    margin: 1,
  },
})

export default House
