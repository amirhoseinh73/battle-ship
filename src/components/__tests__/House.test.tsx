import React from "react"
import renderer from "react-test-renderer"
import House from "../House"

test("renders correctly House", () => {
  const emptyHouse = renderer.create(<House fillType={"blank"} id="1-1" isShip={false} />).toJSON()
  const tinyShipHouse = renderer.create(<House fillType={"tiny"} id="1-2" isShip={true} />).toJSON()

  expect(emptyHouse).toMatchSnapshot()
  expect(tinyShipHouse).toMatchSnapshot()
})
