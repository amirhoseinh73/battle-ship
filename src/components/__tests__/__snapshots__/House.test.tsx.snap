// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly House 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "aspectRatio": 1,
        "borderRadius": 3,
        "height": 32,
        "margin": 1,
        "width": "9%",
      },
      {
        "backgroundColor": "#888",
      },
    ]
  }
>
  <Text>
    1-1
  </Text>
</View>
`;

exports[`renders correctly House 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "aspectRatio": 1,
        "borderRadius": 3,
        "height": 32,
        "margin": 1,
        "width": "9%",
      },
      {
        "backgroundColor": "#FFFF00",
      },
    ]
  }
>
  <Text>
    1-2
  </Text>
</View>
`;
